repose-aggregator/core/repose-core-api/src/main/java/org/openrepose/core/services/reporting/metrics/SummingMeterFactory.java

```diff
@@ -19,22 +19,20 @@
  */
 package org.openrepose.core.services.reporting.metrics;
 
-import com.yammer.metrics.core.TimerContext;
-
-import java.util.concurrent.TimeUnit;
+import com.codahale.metrics.Meter;
 
 /**
- * Interface to allow different TimerByCategory implementations to be used interchangeably.
- * <p/>
- * Any class which implements this interface is expected to be thread-safe.  The individual yammer Timer class is
- * thread-safe.
- * <p/>
- * These objects should be created by the {@link org.openrepose.core.services.reporting.metrics.impl.MetricsServiceImpl}
- * factory class.
+ * A factory used to create {@link Meter}s. The {@link Meter}s created by this
+ * factory will have their values tracked and summed. The summed value will be
+ * reported as its own {@link Meter}.
+ * <p>
+ * This factory is hierarchical. Child factory names will be prefixed by the name
+ * of their parent factory. The overall structure of all factories is a tree.
  */
-public interface TimerByCategory {
+public interface SummingMeterFactory {
```

Comment:
I'd change the name of this to AggregateMeterFactory, and the implementation to SummingMeterFactory. Method names will have to change to reflect that. Which may mean a change in the name of the above all metric?
Hmm, that starts to make me think we may want to change the api in a more general sense. Maybe you ask us for an aggregate and tell us what type when you ask? I'm mostly thinking about the create child factory method and what we would do if you have one type at one level and another another layer in?
Crap, what happens when you want both types? Summing and this as of yet undefined other type?

You know what, ignore all of this for now. This would require more significant changes, and i don't want to go down that rabbit hole right now.